### PR TITLE
[dynamo] Refactor COMPARE_OP and comparison builtins

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #122067
* #122066
* #122063
* #122060
* #122058
* #122055
* __->__ #122043
* #122039

This removes the duplicate handling of comparison ops between symbolic_convert and bultin and refactors the handling to use the binop infrastructure.  This change regresses overheads a bit, but this is fixed in the next PR.

New test skips are variants of `type(e) is np.ndarray` previously falling back to eager.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv [hunk 1]
```diff
@@ -54,47 +54,47 @@ densenet121,pass,0
 
 
 
-detectron2_fasterrcnn_r_101_c4,pass,51
+detectron2_fasterrcnn_r_101_c4,pass,164
```
[the remark this answers is not in the document]
Comment (jansel) [reply benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv:57]: Yeah, I looked into a few and it is just less falling back to eager

**Review by anijain2305 — APPROVED** [review]

File: benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (anijain2305) [reply benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv:57]: Oh nice.

**Comment by jeanschmidt** [comment]
thankfully reverting failed, as seems that there is a FF already landed https://github.com/pytorch/pytorch/pull/122263

**Pull request closed without merging**